Multitouch Support/Native/VoodooI2CNativeEngine.cpp

```diff
@@ -137,14 +139,32 @@ bool VoodooI2CNativeEngine::isForceClickEnabled() {
         lastForceClickPropertyUpdateTime = now_abs;
         return lastIsForceClickEnabled;
     }
-
+    
     // Blocking reading here takes about 10 microseconds
     OSDictionary* dict = OSDynamicCast(OSDictionary, this->getProperty("MultitouchPreferences", gIOServicePlane, kIORegistryIterateRecursively));
     if (!dict) {
         return lastIsForceClickEnabled;
     }
     if (OSCollectionIterator* i = OSCollectionIterator::withCollection(dict)) {
         while (OSSymbol* key = OSDynamicCast(OSSymbol, i->getNextObject())) {
+            // If tap to click (property "Clicking") is disabled,
+            // then we should forbid force clicking,
+            // because it makes impossible to perform a non-force click
+            if (key->isEqualTo("Clicking")) {
+                OSBoolean *boolValue;
+                OSNumber *numValue;
+                if ((boolValue = OSDynamicCast(OSBoolean, dict->getObject(key))) != NULL) {
+                    if (!boolValue->getValue()) {
+                        lastIsForceClickEnabled = FALSE;
+                        break;
+                    }
+                }
+                // is it needed? in default properties it's number, not boolean
```

Comment:
Really Apple? :(

Reply:
Default preferences:
<img width="1043" alt="Знімок екрана 2022-05-13 о 12 39 02" src="https://user-images.githubusercontent.com/446776/168256928-8515e504-fed3-419e-9271-2f5b55f24f88.png">

Current preferences:
<img width="1043" alt="Знімок екрана 2022-05-13 о 12 39 56" src="https://user-images.githubusercontent.com/446776/168256995-35641d2a-0fac-4395-b472-47cf596e21c1.png">

Reply:
If that is the case, can we safely presume that `lastIsForceClickEnabled` should be `false` if the property is not a boolean value? so that we can simplify this a bit as there is no need to read out the number value.